.github/workflows/pull-request.yaml

```diff
@@ -1,16 +1,141 @@
-name: Pull Requests
+name: Build/Test
 
 on:
+  workflow_call:
+    secrets:
+      CHARMHUB_TOKEN:
+        required: true
   pull_request:
     branches:
       - main
 
 jobs:
-  pull-request:
-    name: PR
-    uses: canonical/observability/.github/workflows/charm-pull-request.yaml@main
-    secrets: inherit
-
-  terraform-checks:
-    name: Terraform
-    uses: canonical/observability/.github/workflows/terraform-quality-checks.yaml@main
\ No newline at end of file
+  lint:
+    name: Lint
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+      - name: Install dependencies
+        run: sudo snap install --classic astral-uv
+      - name: Run linters
+        run: make lint
+
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+      - name: Install dependencies
+        run: sudo snap install --classic astral-uv
+      - name: Run tests
+        run: make unit
+
+  lib-check:
+    name: Check libraries
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          fetch-depth: 0
+      - name: Check libs
+        uses: canonical/charming-actions/check-libraries@934193396735701141a1decc3613818e412da606 # 2.6.3
+        with:
+          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+  pack-charm:
+    name: Build charm
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@main
+
+      - name: Install charmcraft
+        run: sudo snap install charmcraft --classic
+
+      - name: Pack charm
+        run: charmcraft pack -v
+
+      - name: Upload charm artifact
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4
+        with:
+          name: charm
+          path: ./*.charm
+
+      - name: Pack tester
+        run: charmcraft pack -v -p ./tests/integration/tester/
+
+      - name: Upload charm artifact
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4
+        with:
+          name: tester-charm
+          path: ./tests/integration/tester/*.charm
```

Comment:
I think the charm will be packed in the same directory as the non-tester one i.e: `./*.charm`